### ISSUE TITLE
"Prompt with Multi-Choice" Node is not able to set property

### ISSUE THREAD

**Issue by tamasschmolcz** [microsoft/BotFramework-Composer#8529]
## Describe the bug

I discovered a weird bug/behaviour. When the user selects e.g. the value "KFZ Zulassung", composer is not able to follow the appropriated path in the Switch Branch. It repeats infinitely asking for input. If I change this value to different random name in the menu prompt + in the Switch branch - I can access the path. Strangely the selection for e.g. the menu value "Wunschkennzeichen reservieren" works perfectly. 

**not working**

![Animation](https://user-images.githubusercontent.com/48097252/128410824-781affe7-fa41-472d-acf8-b81a4af73ffe.gif)

**working after renaming**

![Animation2](https://user-images.githubusercontent.com/48097252/128411983-32df1e01-2a57-4755-a9bf-5260f3432b8a.gif)

**Dialog**

![image](https://user-images.githubusercontent.com/48097252/128413825-51d7a894-f30f-4514-955c-aa629c4137c8.png)

## Version

v2.1.0

## Browser
- [ ] Edge

## OS

- [ ] Windows

## Expected behavior

Multi-choice choosen value is set and the appropriated path is followed.

**Comment by gabog** [comment]
HI @stevkan, any updates on this one?

Thanks

**Comment by stevkan** [comment]
I attempted a repro of the above issue, however I am running into a separate issue where the skill's manifest is not validating while trying to add a skill to the PVA bot.

@tonyanziano, per our discussion, here is the error I am receiving. I've included the manifest below. Please forward to the PVA contact for suggestions. Thanks!

![image](https://user-images.githubusercontent.com/18403946/128905653-f6566318-ff56-4cbd-908b-82c40c0b0b12.png)

```js
{
  "$schema": "https://schemas.botframework.com/schemas/skills/v2.2/skill-manifest.json",
  "$id": "Empty_Bot_0-c91756de-xxxx-xxxx-xxxx-59c804878d2c",
  "endpoints": [
    {
      "protocol": "BotFrameworkV3",
      "name": "x-xxxxx-empty-bot",
      "endpointUrl": "https://x-xxxxx-empty-bot.azurewebsites.net/api/messages",
      "description": "<description>",
      "msAppId": "11c16026-xxxx-xxxx-xxxx-32e7fbd1e33d"
    }
  ],
  "name": "Empty_Bot_0",
  "version": "1.0",
  "publisherName": "Steven K.",
  "activities": {
    "Empty_Bot_0": {
      "type": "event",
      "name": "Empty_Bot_0"
    },
    "AttachmentDialog": {
      "type": "event",
      "name": "AttachmentDialog"
    },
    "Switch_Dialog": {
      "type": "event",
      "name": "Switch_Dialog"
    },
    "conversationUpdate": {
      "type": "conversationUpdate"
    },
    "message": {
      "type": "message"
    },
    "typing": {
      "type": "typing"
    }
  },
  "dispatchModels": {
    "languages": {
      "en-us": [
        {
          "name": "Empty_Bot_0",
          "contentType": "application/lu",
          "url": "https://x-xxxxx-empty-bot.azurewebsites.net/manifests/skill-Empty_Bot_0.en-us.lu",
          "description": "<description>"
        }
      ]
    },
    "intents": [
      "AskForAttachment",
      "Switch_trigger"
    ]
  }
}
```

**Comment by tonyanziano** [comment]
I have verified that in the vanilla Composer runtime (No PVA), setting a property via a Multi-Choice input works as expected:

![image](https://user-images.githubusercontent.com/3452012/128940383-e5685952-7612-4bff-8d01-c75925d23865.png)

This also works in my PVA bot with the same flow:

![image](https://user-images.githubusercontent.com/3452012/128940576-0f64b455-7c64-46e6-ad4f-1c7656620622.png)

@tamasschmolcz maybe double check the details of your Multi-choice prompt and make sure there is nothing weird going on with setting the property.

Here is what mine looks like:

![image](https://user-images.githubusercontent.com/3452012/128940749-e3811a6c-3d89-435a-b78c-a9807f520dc3.png)

**Comment by stevkan** [comment]
@tamasschmolcz, can you double check that you have your `user.selectedOption` property in both the "Property" field under "User input" as well as in the "Condition" field in the "Branch: Switch"?

I tested this playing with a variety of configurations to understand what might be happening. I am able to repro your issue when I place the `user.selectedOption` property in "Value" instead of "Property" under "User input".

![image](https://user-images.githubusercontent.com/18403946/129115788-360625eb-603c-45e2-aba7-46d327bb7d0e.png)

![image](https://user-images.githubusercontent.com/18403946/129115803-be1dd4cd-b00d-4dfd-8783-d8bdc770bb77.png)

**Comment by tamasschmolcz** [comment]
@tonyanziano  @stevkan Thanks guys for taking the time helping me. It is set to the right field "property".  The weird thing is, I even re-built the Prompt and still had the issue.  As I needed to progress with my project I finally renamed it from "Zulassung beantragen" to "KFZ Zulassung beantragen". So it works now.  

Still I have now idea, what's the root cause, but will keep an eye for such a scenario in futur dialogs.
